### DISCUSSION TITLE
Remove the support for editor?

### DISCUSSION THREAD

**Discussion by junsikshim** [maxGraph/maxGraph#79] [Development]
I think we went over this issue before, but I can't remember what the conclusion was, so here we are.
There are 4 files under /src/editor, and if we are removing the support for editor, I'd like to remove them altogether.
Let me know what you think!

@mcyph @tbouffard

**Comment by mcyph** [comment]
I can't think of much reason I would use the editor directly myself, but I think I've basically finished porting it and didn't think it was likely to be much work to maintain it for the time being. 

There were two main reasons I didn't remove it - one was the editor examples were some of the most complex ones, and I thought it would be one of the best test cases to make sure the library functions pretty much the way it used to and I didn't introduce bugs. The other is that I thought there may be some older users of the library which could be using it, and it could ease porting to the new version. I think diagrams.net/draw.io uses it, so it could be useful if anyone wanted to port that to TypeScript later on.

The editor example was the last major one I was planning to finish off before I was able to reasonably confidently declare the library as stable (or at least beta quality), so I was of the view that it could stay in the library for now, and consider removing it later if it becomes an issue for maintenance.

(As a side note, I tried enabling tree shaking by adding `"sideEffects": true` to the package.json file so that files like the editor ones wouldn't be included if people weren't using them, but it caused issues when running the storybook example - I was planning to revisit that when I got back to things but thought should worry about that once the examples were working to make sure that that wasn't the cause of the bugs itself!)

**Comment by tbouffard** [comment]
I agree with @mcyph: the editor examples are the more complex, help checking non regression. I am not using them directly myself, and I don't know if they were widely used by old mxGraph users. Without more information, we should keep them for now.
The editor examples also provide extra features like zoom/panning via CSS transform, custom shapes and styles implementation examples, toolbar usage with actions xml files and more.

We could move the editor code out of the core, in a dedicated 'editor' package to make the separation more visible.
They are also some features that are probably only needed by the editor like translation, function eval when defined inline in actions xml files (this was the way it was done in mxGraph, I don't know if this still work in maxGraph)

**Comment by junsikshim** [comment]
I see. I was trying to change all the string-typed 'styles' into an object one, and there were some style-related utility functions that were only used by the editor example.

I like the idea about the editor example being a separate package, maybe I'll work on it after getting all the styles converted.
